zilliqa/src/state.rs

```diff
@@ -27,7 +27,7 @@ use crate::{
 /// all the keys are hashed and stored in the same sled tree
 pub struct State {
     db: Arc<TrieStorage>,
-    accounts: PatriciaTrie<TrieStorage>,
+    accounts: Arc<Mutex<PatriciaTrie<TrieStorage>>>,
```

Comment:
This feels wrong, given `PatriciaTrie` is basically just `root_hash: B256` + `Arc<TrieStorage>` + some caches for mutations. It is already cheaply cloneable, so why do we need the extra layer?

Also, where is the `Mutex` used? When do we need multiple people to have mutable access to the same storage?